packages/react-select/src/Select.js

```diff
@@ -1463,7 +1463,7 @@ export default class Select extends Component<Props, State> {
             }}
             isFocused={isOptionFocused}
             isDisabled={isDisabled}
-            key={`${this.getOptionValue(opt)}${index}`}
+            key={this.getOptionValue(opt)}
```

Comment:
For reference, this was added to fix https://github.com/JedWatson/react-select/pull/4154.